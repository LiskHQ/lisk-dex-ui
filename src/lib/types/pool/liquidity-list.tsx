export interface LiuqidityListInterface {
    token1Id: string;
    token2Id: string;
    feeTier: string;
    pooledtoken1: string;
    pooledtoken2: string;
    poolTokens: string;
    poolShare: string;
    accumulatedFeesToken1: string;
    accumulatedFeesToken2: string;
    totalFees: string;
  }