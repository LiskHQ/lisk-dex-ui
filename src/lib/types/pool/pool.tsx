export interface LiquidityListInterface {
  token1Id: string;
  token2Id: string;
  feeTier: string;
  pooledtoken1: string;
  pooledtoken2: string;
  poolTokens: string;
  poolShare: string;
  accumulatedFeesToken1: string;
  accumulatedFeesToken2: string;
  totalFees: string;
}

export interface IpriceRange {
  minPrice: string,
  maxPrice: string,
  currentPrice: string  
}

export  interface IdepositAmounts {
  token1: string,
  token2: string,
  amount1: string,
  amount2: string,
  feeTier: string,
}
