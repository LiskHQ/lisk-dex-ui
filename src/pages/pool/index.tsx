import { NextPage } from 'next';
import { PoolContainer } from 'containers';
import { withLayout } from 'components';

const PoolPage: NextPage = () => {
  return (
    <PoolContainer />
  );
};

export default withLayout(PoolPage);