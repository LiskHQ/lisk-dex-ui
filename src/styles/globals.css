:root {
  font-size: 16px;
}

html,
body,
#__next {
  padding: 0 !important;
  margin: 0;
  height: 100%;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

a,
.btn,
button,
span,
p,
i,
input,
select,
textarea,
li,
img,
svg path,
*::after,
*::before,
.transition-3,
h1,
h2,
h3,
h4,
h5,
h6 {
  transition: all 0.3s ease-out 0s;
}

span,
p,
a,
li.MuiMenuItem-root,
label,
label.MuiFormLabel-root,
span.MuiTypography-root,
p.MuiTypography-root,
button.MuiButton-root,
button.MuiButtonBase-root,
.MuiInput-root,
.MuiTableCell-root {
  font-family: 'Roboto', sans-serif;
}

h1,
h2,
h3,
h4,
h5,
h6,
h1.MuiTypography-root,
h2.MuiTypography-root,
h3.MuiTypography-root,
h4.MuiTypography-root,
h5.MuiTypography-root,
h6.MuiTypography-root {
  font-family: 'Roboto', sans-serif;
}