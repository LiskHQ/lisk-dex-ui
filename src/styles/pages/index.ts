import { GovernanceStyle } from "./governance/index.style";

export {
  GovernanceStyle
}