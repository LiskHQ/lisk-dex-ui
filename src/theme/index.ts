export { default as theme } from "./theme";
export { default as variables } from "./variables";
