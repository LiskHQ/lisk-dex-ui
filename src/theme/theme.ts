import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
