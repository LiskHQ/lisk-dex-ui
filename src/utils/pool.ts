export const getPoolToken0 = (pool: string) => {
  return pool.split('-')[0];
};

export const getPoolToken1 = (pool: string) => {
  return pool.split('-')[1];
};