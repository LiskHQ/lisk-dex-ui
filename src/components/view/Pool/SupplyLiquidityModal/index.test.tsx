import React from 'react';
import { ThemeProvider } from '@mui/material';
import { fireEvent, render } from '@testing-library/react';
import { SupplyLiquidityModal, ISupplyLiquidityModalProps } from './index';
import { lightTheme } from 'styles/theme';
import { mockPool } from '__mock__/pool.mock';

function renderComponent(props: ISupplyLiquidityModalProps) {
  return render(
    <ThemeProvider theme={lightTheme} >
      <SupplyLiquidityModal {...props} />
    </ThemeProvider>
  );
}

describe('SupplyLiquidityModal component', () => {
  const mockProps: ISupplyLiquidityModalProps = {
    pool: mockPool,
    onClose: jest.fn(),
    onConfirm: jest.fn(),
  };

  it('checks if the component matches the snapshot', () => {
    const { container } = renderComponent(mockProps);
    expect(container).toMatchSnapshot();
  });

  it('confirm click work', () => {
    const { getByTestId } = renderComponent(mockProps);
    fireEvent.click(getByTestId('supply-liquidity-modal-button-test'));

    expect(mockProps.onConfirm).toBeCalled();
  });

  it('cancel click work', () => {
    const { getByTestId } = renderComponent(mockProps);
    fireEvent.click(getByTestId('supply-liquidity-modal-cancel-test'));

    expect(mockProps.onClose).toBeCalled();
  });
});
