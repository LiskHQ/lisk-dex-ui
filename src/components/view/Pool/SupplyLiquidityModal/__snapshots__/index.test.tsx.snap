// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplyLiquidityModal component checks if the component matches the snapshot 1`] = `
.emotion-0 .supply-liquidity-modal-background {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.5);
  z-index: 30;
}

.emotion-0 .supply-liquidity-modal-container {
  position: fixed;
  left: 50%;
  top: 50%;
  margin-bottom: 10rem;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  width: 33.25rem;
  border-radius: 0.5rem;
  background: #FFFFFF!important;
  box-shadow: 0px 1px 7px rgba(0, 0, 0, 0.1);
  z-index: 30;
}

@media (max-width:575.95px) {
  .emotion-0 .supply-liquidity-modal-container {
    width: 100%;
  }
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header {
  padding: 2rem 1.5rem;
  position: relative;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .supply-liquidity-modal-title {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #282828;
  margin-bottom: 4rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .supply-liquidity-modal-title .MuiIconButton-root {
  position: absolute;
  top: 1.5rem;
  right: 1.5rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .supply-liquidity-modal-title .MuiIconButton-root svg path {
  stroke: #282828;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .supply-liquidity-modal-description {
  margin-bottom: 1.25rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .supply-liquidity-modal-description div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-bottom: 2rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .supply-liquidity-modal-description p {
  text-align: center;
  margin-bottom: 0.75rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .light-icon {
  position: absolute;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .light-icon.data-index-1 {
  top: 10rem;
  left: 10rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .light-icon.data-index-2 {
  top: 2rem;
  left: 26rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .light-icon.data-index-3 {
  top: 4rem;
  left: 2rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .light-icon.data-index-4 {
  top: 8rem;
  left: 22rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .light-icon.data-index-5 {
  top: 16rem;
  left: 6rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-header .light-icon.data-index-6 {
  top: 14rem;
  left: 28rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-body {
  padding: 1.5rem 1.5rem 2rem 1.5rem;
  border-radius: 0 0 0.5rem 0.5rem;
  background: #F2F5F9;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-body .deposit-property {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 0.75rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-body .deposit-property .deposit-property-amount {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-body .deposit-property .deposit-property-amount p {
  margin-left: 0.5rem;
}

.emotion-0 .supply-liquidity-modal-container .supply-liquidity-modal-body .deposit-description {
  margin: 3.25rem 2rem 2rem 2rem;
  text-align: center;
}

.emotion-5 {
  margin: 0;
  font-size: 1.25rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00735em;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-6::-moz-focus-inner {
  border-style: none;
}

.emotion-6.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-6 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-6:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-6:hover {
    background-color: transparent;
  }
}

.emotion-6.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-10 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.375rem;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
}

.emotion-25 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  width: 100%;
  color: #FF4557;
  padding: 0.75rem;
  text-transform: capitalize;
}

.emotion-25::-moz-focus-inner {
  border-style: none;
}

.emotion-25.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-25 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-25:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-25:hover {
    background-color: #1976d2;
  }
}

.emotion-25:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-25.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-25.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-25.MuiButton-outlined {
  border-color: #FF4557;
}

.emotion-25.MuiButton-contained {
  background: #6953F4;
  color: #FFFFFF;
}

.emotion-25.Mui-disabled {
  opacity: 0.7;
}

<div>
  <div
    class="emotion-0"
    data-testid="supply-liquidity-modal-test"
  >
    <div
      class="supply-liquidity-modal-background MuiBox-root emotion-1"
    />
    <div
      class="supply-liquidity-modal-container MuiBox-root emotion-1"
    >
      <div
        class="supply-liquidity-modal-header MuiBox-root emotion-1"
      >
        <svg
          class="light-icon data-index-1"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 23.154L1.698 21.456L2.544 22.302L0.846 24H0V23.154ZM0 0.84L1.698 2.538L2.544 1.692L0.846 0H0V0.846V0.84ZM23.154 24L21.456 22.302L22.302 21.456L24 23.154V24H23.154ZM24 0.846L22.302 2.544L21.456 1.698L23.154 0H24V0.846ZM12 11.16L13.698 9.462L14.544 10.308L12.846 12L14.544 13.698L13.698 14.544L12 12.846L10.302 14.544L9.456 13.698L11.154 12L9.456 10.302L10.302 9.456L12 11.154V11.16Z"
            fill="#C2C2EF"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          class="light-icon data-index-2"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 23.154L1.698 21.456L2.544 22.302L0.846 24H0V23.154ZM0 0.84L1.698 2.538L2.544 1.692L0.846 0H0V0.846V0.84ZM23.154 24L21.456 22.302L22.302 21.456L24 23.154V24H23.154ZM24 0.846L22.302 2.544L21.456 1.698L23.154 0H24V0.846ZM12 11.16L13.698 9.462L14.544 10.308L12.846 12L14.544 13.698L13.698 14.544L12 12.846L10.302 14.544L9.456 13.698L11.154 12L9.456 10.302L10.302 9.456L12 11.154V11.16Z"
            fill="#C2C2EF"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          class="light-icon data-index-3"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 23.154L1.698 21.456L2.544 22.302L0.846 24H0V23.154ZM0 0.84L1.698 2.538L2.544 1.692L0.846 0H0V0.846V0.84ZM23.154 24L21.456 22.302L22.302 21.456L24 23.154V24H23.154ZM24 0.846L22.302 2.544L21.456 1.698L23.154 0H24V0.846ZM12 11.16L13.698 9.462L14.544 10.308L12.846 12L14.544 13.698L13.698 14.544L12 12.846L10.302 14.544L9.456 13.698L11.154 12L9.456 10.302L10.302 9.456L12 11.154V11.16Z"
            fill="#C2C2EF"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          class="light-icon data-index-4"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 23.154L1.698 21.456L2.544 22.302L0.846 24H0V23.154ZM0 0.84L1.698 2.538L2.544 1.692L0.846 0H0V0.846V0.84ZM23.154 24L21.456 22.302L22.302 21.456L24 23.154V24H23.154ZM24 0.846L22.302 2.544L21.456 1.698L23.154 0H24V0.846ZM12 11.16L13.698 9.462L14.544 10.308L12.846 12L14.544 13.698L13.698 14.544L12 12.846L10.302 14.544L9.456 13.698L11.154 12L9.456 10.302L10.302 9.456L12 11.154V11.16Z"
            fill="#C2C2EF"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          class="light-icon data-index-5"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 23.154L1.698 21.456L2.544 22.302L0.846 24H0V23.154ZM0 0.84L1.698 2.538L2.544 1.692L0.846 0H0V0.846V0.84ZM23.154 24L21.456 22.302L22.302 21.456L24 23.154V24H23.154ZM24 0.846L22.302 2.544L21.456 1.698L23.154 0H24V0.846ZM12 11.16L13.698 9.462L14.544 10.308L12.846 12L14.544 13.698L13.698 14.544L12 12.846L10.302 14.544L9.456 13.698L11.154 12L9.456 10.302L10.302 9.456L12 11.154V11.16Z"
            fill="#C2C2EF"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          class="light-icon data-index-6"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 23.154L1.698 21.456L2.544 22.302L0.846 24H0V23.154ZM0 0.84L1.698 2.538L2.544 1.692L0.846 0H0V0.846V0.84ZM23.154 24L21.456 22.302L22.302 21.456L24 23.154V24H23.154ZM24 0.846L22.302 2.544L21.456 1.698L23.154 0H24V0.846ZM12 11.16L13.698 9.462L14.544 10.308L12.846 12L14.544 13.698L13.698 14.544L12 12.846L10.302 14.544L9.456 13.698L11.154 12L9.456 10.302L10.302 9.456L12 11.154V11.16Z"
            fill="#C2C2EF"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="supply-liquidity-modal-title MuiBox-root emotion-1"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 emotion-5"
          >
            Supplying Liquidity
          </h4>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-6"
            data-testid="supply-liquidity-modal-cancel-test"
            tabindex="0"
            type="button"
          >
            <svg
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_625_48983)"
              >
                <path
                  d="M15 5L5 15"
                  stroke="#282828"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.66667"
                />
                <path
                  d="M5 5L15 15"
                  stroke="#282828"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.66667"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_625_48983"
                >
                  <rect
                    fill="white"
                    height="20"
                    width="20"
                  />
                </clippath>
              </defs>
            </svg>
            <span
              class="MuiTouchRipple-root emotion-7"
            />
          </button>
        </div>
        <div
          class="supply-liquidity-modal-description MuiBox-root emotion-1"
        >
          <div
            class="MuiBox-root emotion-1"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2748%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2748%27%20height=%2748%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            Supplying 
            1862
             
            LSK
             and 
            0.045
             
            ETH
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            You will receive 3.45 
            LSK
            /
            ETH
             LP Tokens
          </p>
        </div>
      </div>
      <div
        class="supply-liquidity-modal-body MuiBox-root emotion-1"
      >
        <div
          class="deposit-property MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            LSK
             Deposited:
          </p>
          <div
            class="deposit-property-amount MuiBox-root emotion-1"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body1 emotion-10"
            >
              1862
            </p>
          </div>
        </div>
        <div
          class="deposit-property MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            ETH
             Deposited:
          </p>
          <div
            class="deposit-property-amount MuiBox-root emotion-1"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body1 emotion-10"
            >
              0.045
            </p>
          </div>
        </div>
        <div
          class="deposit-property MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            Share of pool:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            0.09%
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 deposit-description emotion-10"
        >
          Output is estimated. If the price changes by more than 0.5% your transaction will revert.
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium supply-liquidity-modal-confirm emotion-25"
          data-testid="supply-liquidity-modal-button-test"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            Go to Confirm
          </p>
          <span
            class="MuiTouchRipple-root emotion-7"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
